src/sources/syslog.rs

```diff
@@ -247,9 +278,43 @@ fn event_from_str(host_key: &String, host: Option<Bytes>, raw: impl AsRef<str>)
 
             event
         })
+        .map_err(|_| warn!("Problem parsing incoming message, check syslog format"))
         .ok()
 }
 
+fn create_event_from_rfc5424(parsed: &SyslogMessage) -> Event {
+    let mut event = Event::from(&parsed.msg[..]);
+    let log = event.as_mut_log();
+
+    log.insert_implicit("severity".into(), parsed.severity.as_str().into());
+    log.insert_implicit("facility".into(), parsed.facility.as_str().into());
+    log.insert_implicit("version".into(), parsed.version.into());
+
+    if let Some(app_name) = parsed.appname.clone() {
+        log.insert_implicit("appname".into(), app_name.into());
+    }
+    if let Some(msg_id) = parsed.msgid.clone() {
```

Comment:
The clone here and the one for `appname` should be able to go away if we change this function to take `parsed` by value instead of reference.

Reply:
Goog point